tests/integration/dynamic/python-secrets-separate-module/simple_provider.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2016-2024, Pulumi Corporation.  All rights reserved.
+
+import pulumi
+from pulumi.dynamic import CreateResult, Resource, ResourceProvider
+
+
+config = pulumi.Config()
+password = config.require_secret("password")
```

Comment:
The test currently fails with:

```
error: Exception calling application: Missing required configuration variable 'project:password'
	please set a value using the command `pulumi config set --secret project:password <value>`
```

I haven't debugged it, but I believe this is because the project name isn't being setup for the dynamic provider (`pulumi.Config()` is shorthand for `pulumi.Config("the-project-name")`.

If I do specify the project name explicitly, e.g.

```python
config = pulumi.Config("dynamic-provider-secret-separate-module-py")
```

The test fails with a different error:

```
error: Exception calling application: There is no current event loop in thread 'ThreadPoolExecutor-0_0'.
```

I haven't dug into why.